docker/daemon/extra/store.go

```diff
@@ -0,0 +1,130 @@
+package extra
+
+// extra.Store allows recording metadata (manifests and signatures) which make it possible to authenticate docker images even if the daemon does not store this data itself.
+// This is implemented by storing data on the filesystem:
+// ${root}/${config_digest}/@${manifest_digest} is a subdirectory (see dir.go)
+// ${root}/${config_digest}/${edited_name}:${tag} is a symlink to @${manifest_digest} in the same directory (there may be multiple symlinks pointing at the same manifest digest)
+//     ${edited_name} is ${name} with '/' replaced by ','
+
+import (
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/containers/image/docker/reference"
+	"github.com/containers/image/manifest"
+	"github.com/containers/image/types"
+	"github.com/opencontainers/go-digest"
+)
+
+// systemDockerExtraDirPath is the path to the docker extra storage, used for recording metadata allowing to authenticate docker images.
+// You can override this at build time with
+// -ldflags '-X github.com/containers/image/docker/daemon/extra.systemDockerExtraDirPath=$your_path'
+var systemDockerExtraDirPath = builtinDockerExtraDirPath
```

Comment:
I still believe compile-time changes aren't good for this kind of stuff. Ideally, go codes have a way to pass a `Ctx` struct down the function calls, that `Ctx` should contain this information.